### PR TITLE
Add Cantonese Openrice Classification task

### PR DESCRIPTION
**Pull request by izhx**
## Checklist for adding MMTEB dataset

Reason for dataset addition: cover Cantonese

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] I have considered the size of the dataset and reduced it if it is too big (2048 examples is typically large enough for most tasks)
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [POINTS.md](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) file.

### Additional results
I run the task by a Cantonese pretrained model [`toastynews/electra-hongkongese-small-discriminator`](https://huggingface.co/toastynews/electra-hongkongese-small-discriminator) (base model, not embedding model) and get much higher scores (`0.40`) than `mE5` and `paraphrase-multilingual-MiniLM-L12-v2` (`~= 0.30`).

```
{
  "dataset_revision": "b8b9a3d0e40bc0674945a0edb0bc549b554d6f20",
  "mteb_dataset_name": "YueOpenriceClassification",
  "mteb_version": "1.6.3",
  "test": {
    "accuracy": 0.4001298490504788,
    "accuracy_stderr": 0.01464472655189897,
    "evaluation_time": 14.6,
    "f1": 0.36628574491919214,
    "f1_stderr": 0.013082359119775245,
    "main_score": 0.4001298490504788
  }
}
```

### For tests
My test failed due to network reasons (of China...).

```
===================================================================== slowest 5 durations =====================================================================
43.37s call     tests/test_mteb.py::test_mteb_task[average_word_embeddings_levy_dependency-NFCorpus]
32.41s call     tests/test_all_abstasks.py::test_dataset_availability
29.73s call     tests/test_mteb.py::test_mteb_task[average_word_embeddings_levy_dependency-TwentyNewsgroupsClustering]
27.88s call     tests/test_mteb.py::test_two_mteb_tasks
27.23s call     tests/test_mteb.py::test_mteb_task[average_word_embeddings_levy_dependency-SprintDuplicateQuestions]
=================================================================== short test summary info ===================================================================
FAILED tests/test_all_abstasks.py::test_dataset_availability - aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host huggingface.co:443 ssl:default [Connect call failed ('104.244.43.136', 443)]
=============================================== 1 failed, 149 passed, 81 skipped, 1 warning in 78.09s (0:01:18) ===============================================
make: *** [Makefile:17: test] Error 1
```

### CODE REVIEW

File: mteb/tasks/Classification/yue/YueOpenriceClassification.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        eval_langs={"yue": ["yue-Hant"]},
+        main_score="accuracy",
+        date=None,
+        form=["spoken"],
+        domains=["Reviews"],
+        task_subtypes=["Sentiment/Hate speech"],
+        license=None,
+        socioeconomic_status=None,
+        annotations_creators="human-annotated",
+        dialect=None,
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/yue/YueOpenriceClassification.py:28]:
```suggestion
        dialect=[],
```
Assuming there are no dialects?

File: mteb/tasks/Classification/yue/YueOpenriceClassification.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        type="Classification",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs={"yue": ["yue-Hant"]},
+        main_score="accuracy",
+        date=None,
+        form=["spoken"],
+        domains=["Reviews"],
+        task_subtypes=["Sentiment/Hate speech"],
+        license=None,
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/yue/YueOpenriceClassification.py:25]: Required, if none is specified then use "Not specified"

File: mteb/tasks/Classification/yue/YueOpenriceClassification.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        dataset={
+            "path": "izhx/yue-openrice-senti",
+            "revision": "b8b9a3d0e40bc0674945a0edb0bc549b554d6f20",
+        },
+        type="Classification",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs={"yue": ["yue-Hant"]},
+        main_score="accuracy",
+        date=None,
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/yue/YueOpenriceClassification.py:21]: Time period for when the data was created was annotated as a tuple (from, to). Can be a guess.

File: mteb/tasks/Classification/yue/YueOpenriceClassification.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        form=["spoken"],
+        domains=["Reviews"],
+        task_subtypes=["Sentiment/Hate speech"],
+        license=None,
+        socioeconomic_status=None,
```
Comment (KennethEnevoldsen) [inline mteb/tasks/Classification/yue/YueOpenriceClassification.py:26]:
```suggestion
        socioeconomic_status="mixed",
```
Assuming as it is reviews

**Comment by KennethEnevoldsen** [comment]
Thanks for the additional results, seems like the multilingual models indeed are lacking.

File: mteb/tasks/Classification/yue/YueOpenriceClassification.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
```
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply mteb/tasks/Classification/yue/YueOpenriceClassification.py]: I would move it to the `zho` folder as well (or potentially split the `zho` folder into multiple folders (the first one causes less issues so I would go for than one)

**Review by imenelydiaker — APPROVED** [review]
I think it's good for merging, @KennethEnevoldsen wdyt?

**Comment by imenelydiaker** [comment]
Also, can you please add your contribution points to the POINTS.md file?

**Comment by KennethEnevoldsen** [comment]
> I think it's good for merging, @KennethEnevoldsen wdyt?

I think this PR looks great! Thanks for the contribution @izhx

**Pull request merged**